controller/array_action/array_mediator_xiv.py

```diff
@@ -114,6 +114,13 @@ def is_active(self):
     def _convert_size_blocks_to_bytes(self, size_in_blocks):
         return int(size_in_blocks) * self.BLOCK_SIZE_IN_BYTES
 
+    def _get_copy_source_object_wwn(self, cli_object):
+        if hasattr(cli_object, "copy_master_wwn"):
```

Comment:
When it doesn't have this attribute?

Reply:
on gen3...

Reply:
The code isn't clear enough. I suggest clarifying it by one of:
1. Create a gen3 mediator that inherits from this class and override the relevant function
2. Divide the function into two functions (one for gen3 and one for other gens)

WDYT?

Reply:
a bit overkill IMO and the code looks pretty clear to me.
we don't even officially support gen3, it's mainly for our tests

Reply:
If we don't officially support gen3, then it's even more important to explain why did we do so in the code.
Currently, nothing in this function implies gen3.

Reply:
would you like me to copy [these ](https://github.com/IBM/ibm-block-csi-driver/blob/793cb5b37e1f1e24900b5fd84a4f3c4572bd9a51/controller/array_action/array_mediator_xiv.py#L118)messages to this method too?

Reply:
I don't think we should maintain comments.
if we are certain that this scenario is relevant to (and only to) gen3, then we can convey this info by code, e.g. wrap the `hasattr` in a `is_gen3` method

Reply:
or `is_not_gen3`..

Reply:
not sure why `not` is better

Reply:
cause if its `hasattr` is true so its not gen3..

Reply:
The function can `return not hasattr...`

Reply:
then it might be better for the `not` to be with `hasattr` - @zingero?

Reply:
so what decided?

Reply:
`return not hasattr` sounds good to me

Reply:
done